freqtrade/exchange/exchange_utils.py

```diff
@@ -219,35 +220,48 @@ def amount_to_contract_precision(
     return amount
 
 
-def price_to_precision(price: float, price_precision: Optional[float],
-                       precisionMode: Optional[int]) -> float:
+def price_to_precision(
+    price: float,
+    price_precision: Optional[float],
+    precisionMode: Optional[int],
+    rounding_mode: int = ROUND_UP,
+) -> float:
     """
     Returns the price rounded up to the precision the Exchange accepts.
     Partial Re-implementation of ccxt internal method decimal_to_precision(),
     which does not support rounding up
     TODO: If ccxt supports ROUND_UP for decimal_to_precision(), we could remove this and
     align with amount_to_precision().
-    !!! Rounds up
     :param price: price to convert
     :param price_precision: price precision to use. Used from markets[pair]['precision']['price']
     :param precisionMode: precision mode to use. Should be used from precisionMode
                           one of ccxt's DECIMAL_PLACES, SIGNIFICANT_DIGITS, or TICK_SIZE
+    :param rounding_mode: rounding mode to use. Defaults to ROUND_UP
     :return: price rounded up to the precision the Exchange accepts
-
     """
     if price_precision is not None and precisionMode is not None:
-        # price = float(decimal_to_precision(price, rounding_mode=ROUND,
-        #                                    precision=price_precision,
-        #                                    counting_mode=self.precisionMode,
-        #                                    ))
         if precisionMode == TICK_SIZE:
+            if rounding_mode == ROUND:
```

Comment:
using Round (especially with tick-size exchanges) is pretty dangerous - as it doesn't really define the direction of the rounding.

Reply:
Can you please elaborate more on this? Maybe I miss some information on properties of "tick-size" exchanges or how it's used?

Reply:
TICK_SIZE is a rare method - fortunately used by very few exchanges.

You could for example get tick value of 0.5 - which would mean only multiples of 0.5 are valid prices (100, 100.5, 101, ...) - but not 100.1, 100.2, ...). 

As a consequence - using "round" of 100.2 (or 100.4) would ALWAYS round down to 100, while 100.7 (or 100.55) would always round to 101 ... which is however incorrect depending on the usage (which is why I'm in general not a fan of using ROUND).

Depending on the  price of the pair (VERY low priced pairs) - tick-size can also be 1 or 5 - or 2 ... so 2,4,6, ... are valid - but 5,7 is not (obviously decimals would also not be allowed in this mode) - which makes it even less predictable / worse - as the actual difference increases.

Reply:
Oh, so if I understand right, it's dangerous when the TICK_SIZE is not a power of 10 due to its complexity for an average human brain as its not used to the "non-decimal" rounding, and it would be slightly more complex for a human to compute the normal rounding vs a ROUND_UP/ROUND_DOWN, and this is the source of danger.

I don't agree that a slight increase in complexity would be a valid reason to implement a functionality in a semantically wrong way, as this would lead to even more confusion and would make system even more complex and prone to unexpected behavior.

Reply:
independent of tick-size - using ROUND is wrong for most cases (see the other comments).